lua/telescope/pickers/layout_strategies.lua

```diff
@@ -391,6 +391,91 @@ layout_strategies.center = make_documented_layout("center", vim.tbl_extend("erro
   }
 end)
 
+--- Cursor layout dynamically positioned below the cursor if possible.
+--- If there is no place below the cursor it will be placed above.
+---
+--- <pre>
+--- ┌──────────────────────────────────────────────────┐
+--- │                                                  │
+--- │   █                                              │
+--- │   ┌──────────────┐┌─────────────────────┐        │
+--- │   │    Prompt    ││      Preview        │        │
+--- │   ├──────────────┤│      Preview        │        │
+--- │   │    Result    ││      Preview        │        │
+--- │   │    Result    ││      Preview        │        │
+--- │   └──────────────┘└─────────────────────┘        │
+--- │                                         █        │
+--- │                                                  │
+--- │                                                  │
+--- │                                                  │
+--- │                                                  │
+--- │                                                  │
+--- └──────────────────────────────────────────────────┘
+--- </pre>
+layout_strategies.cursor = make_documented_layout("cursor", vim.tbl_extend("error", shared_options, {
+}), function(self, max_columns, max_lines, layout_config)
+  local initial_options = p_window.get_initial_window_options(self)
+  local preview = initial_options.preview
+  local results = initial_options.results
+  local prompt = initial_options.prompt
+
+  local height_opt = layout_config.height
+  local height = resolve.resolve_height(height_opt)(self, max_columns, max_lines)
+
+  local width_opt = layout_config.width
+  local width = resolve.resolve_width(width_opt)(self, max_columns, max_lines)
+
+  local max_width = (width > max_columns and max_columns or width)
+
+  local bs = get_border_size(self)
+
+  prompt.height = 1
+  results.height = height
+  preview.height = results.height + prompt.height + bs
+
+  -- If previewer is activated, it should take 2/3 of the space (width)
+  local width_left = self.previewer and math.floor(max_width / 3) or max_width
+  prompt.width = width_left
+  results.width = prompt.width
+  preview.width = self.previewer and (2*width_left) or 0
```

Comment:
Is there a reason we are hard coding the width of the previewer?
We could include a `preview_width` key in a similar way to `layout_strategies.horizontal`.

Reply:
Do you mean if `self.previewer` is false?
I'm guessing this is just a leftover from testing. I can remove it since it shouldn't affect anything I think?

Reply:
Sorry for the confusion, I mean if `self.previewer` is `true`.

If `self.previewer` is `false`, then yes, `width_left` should be `max_width`.
But if the user wants a previewer and `self.previewer` is `true`, then the width of the previewer should be configurable.
I would suggest looking at the implementation of `preview_width` in `layout_strategies.horizontal` as a guide for how it could be implemented in `layout_strategies.cursor`.

Reply:
Ah yes, I think I understand now, I'm going to look into that!

Reply:
I think I've implemented it correctly now.